File: Behavioral/Strategy/JavaScript/payPalPayment.js
// payPalPayment.js

const PaymentStrategy = require("./paymentStrategy");

class PayPalPayment extends PaymentStrategy {
  constructor(email) {
    super();
    this.email = email;
  }

  pay(amount) {
    console.log(`Paid $${amount} using PayPal (Email: ${this.email})`);
  }
}

module.exports = PayPalPayment;

File: Behavioral/Strategy/Python/paypal_payment.py
# paypal_payment.py

from payment_strategy import PaymentStrategy

class PayPalPayment(PaymentStrategy):
  def __init__(self, email):
    self.email = email

  def pay(self, amount):
    print(f"Paid ${amount} using PayPal (Email: {self.email})")
